### PR TITLE
Correctly compile `lang="sass" styles`

### LINKED ISSUE #851: 🐛 BUG: `lang="sass"` on a <style> uses `scss` istead
### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

Docs give an example of using both `sass` and `scss`
>👓 Sass
>Astro also supports Sass out-of-the-box. To enable for each framework:
 >> Astro: <style lang="scss"> or <style lang="sass">

https://docs.astro.build/guides/styling#frameworks-and-libraries

But Astro uses `scss` to compile in both cases (even if told to use `sass`)

### Steps to Reproduce

1. `npm init astro` using template `Starter Kit (Generic)`
2. Replace `index.astro` with:
```html
<head>
    <style lang="sass">
        span
            color: red
    </style>
</head>
<body>
    <span>Test</span>
</body>
```
3. Run `npm start`
4.  Observe error:
```
Error: expected "{".
  ╷
3 │             color: red
  │                       ^
  ╵
  stdin 3:23  root stylesheet
````

5. Format it like an `scss` stylesheet (keeping lang as `sass`):
```html
<head>
    <style lang="sass">
        span {
            color: red;
        }
    </style>
</head>
<body>
    <span>Test</span>
</body>
```
6. See it working without any issues

### Link to Minimal Reproducible Example (Optional)

_No response_

### PR DESCRIPTION
**Pull request by Tc-001**
close #851 

## Changes

- Adds `indentedSyntax: true` for compiling `sass` stylesheets, so that they are no longer compiled as `scss`

## Testing

Tested by recreating steps in #851 (with locally built Astro), now shows correct behavior

## Docs

Current behavior reflects that described in the documentation, thus no change needed

### CODE REVIEW

**Comment by jasikpark** [comment]
Maybe there should be an update to the docs to indicate that you will get different results from the different lang= values since I believe the docs currently claim both render as sass.

**Comment by Tc-001** [comment]
Documentation kinda already mentions it:
> And the best part? You know it already! Use .scss to write familiar CSS syntax you’re used to, and only sprinkle in Sass features if/when you need them.

And https://sass-lang.com/guide compares both side by side.

But I agree, the docs could be a bit clearer.

**Comment by jasikpark** [comment]
> Documentation kinda already mentions it:
> 
> > And the best part? You know it already! Use .scss to write familiar CSS syntax you’re used to, and only sprinkle in Sass features if/when you need them.
> 
> 
> 
> And https://sass-lang.com/guide compares both side by side.
> 
> 
> 
> But I agree, the docs could be a bit clearer.

That blurb sounds more like it's saying you can write vanilla css in scss and sprinkle in the extra language features as needed. 🤷‍♀️

**Comment by jasikpark** [comment]
I definitely feel like linking to that guide would be good! Maybe a blurb on the end about "sass is like this vs scss is like this" that links to the sass guide that compares them would make it a bit more clear? I wonder if the section title should be `Sass and SCSS` then 🤔

**Comment by Tc-001** [comment]
"Sass and Scss" sound good!

Then after the title it could describe or link to some comparison between the two, plus the "getting started" guide.

File: packages/astro/src/compiler/transform/styles.ts [hunk 1]
```diff
@@ -107,7 +107,10 @@ async function transformStyle(code: string, { logging, type, filename, scopedCla
       css = code;
       break;
     }
-    case 'sass':
+    case 'sass': {
+      css = sass.renderSync({ data: code, includePaths, indentedSyntax: true }).css.toString('utf8');
```
Comment (drwpow) [inline packages/astro/src/compiler/transform/styles.ts:111]: Thanks for adding this! But would you mind keeping the command combined, as it was before?

```js
      css = sass.renderSync({ data: code, includePaths, indentedSyntax: styleType === 'sass' }).css.toString('utf8');
```

Reason is that when separating these, there’s a good chance that someone would accidentally change behavior for one but not the other. But by keeping them combined, we guarantee both flavors will always get the same settings.

**Review by drwpow — CHANGES_REQUESTED** [review]
+1 in favor of this fix, but with a minor RFC as outlined in a comment (above)

File: packages/astro/src/compiler/transform/styles.ts (hunk 1 above)
Comment (Tc-001) [reply packages/astro/src/compiler/transform/styles.ts:111]: Sounds good 👍

**Comment by Tc-001** [comment]
Oh right... can't change from GitHub alone. 
Will add the change when I get home

**Comment by jasikpark** [comment]
here's the link to the md file for the docs that need updating: https://github.com/snowpackjs/astro/blob/main/docs/src/pages/guides/styling.md

File: .changeset/itchy-bobcats-rest.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
```
Comment (drwpow) [inline .changeset/itchy-bobcats-rest.md:2]: Oh another minor change while you’re at it: I think this should be a `patch` not a minor

Comment (Tc-001) [reply .changeset/itchy-bobcats-rest.md:2]: I see... Was a bit confused about that. Iirc `yarn changeset` only gave me an option between major and minor.
Would just editing the file fix it? Or do I need to run a specific command and delete the current file? (I'm not familiar with changesets as a whole, a great opportunity to learn)

File: packages/astro/src/compiler/transform/styles.ts (hunk 1 above)
Comment (Tc-001) [reply packages/astro/src/compiler/transform/styles.ts]: Added in 7852c4a

File: .changeset/itchy-bobcats-rest.md (hunk 2 above)
Comment (Tc-001) [reply .changeset/itchy-bobcats-rest.md]: Added in 7393eac

**Comment by Tc-001** [comment]
Reworded some of the docs, now reads like this:

----

### 👓 Sass and Scss

Astro also supports [Sass](https://sass-lang.com/) (and Scss) out-of-the-box. To enable for each framework:

- **Astro**: `<style lang="sass">`or `<style lang="scss">` 
- **React** / **Preact**: `import Styles from './styles.module.scss'`;
- **Vue**: `<style lang="sass">`or `<style lang="scss">` 
- **Svelte**: `<style lang="sass">`or `<style lang="scss">` 

💁‍ Sass is great! If you haven’t used Sass in a while, here is a quick [refresher](https://sass-lang.com/guide). Also, the new and improved [Sass Modules](https://sass-lang.com/) are a great fit with modern web development. And the best part? **You know it already!** If you are used to the syntax of regular CSS, use `.scss`, and only sprinkle in Sass features if/when you need them.

----

- Added scss to the title and first paragraph
- Moves around so `sass` comes before `scss` everywhere, because the paragraph at the bottom presents `scss` as an alternative. 
- Reworded some of that bottom paragraph (no need for blazing fast if it gets statically generated anyways)

For some reason it reads a bit rushed for me, but may just be me being sleep deprived :sweat_smile: 
Should the docs change be part of this PR too?

**Pull request merged**